CMake configuration fails

### Checklist

- [x] I have searched for [similar issues](https://github.com/isl-org/Open3D/issues).
- [x] For Python issues, I have tested with the [latest development wheel](https://www.open3d.org/docs/latest/getting_started.html#development-version-pip).
- [x] I have checked the [release documentation](https://www.open3d.org/docs/release/) and the [latest documentation](https://www.open3d.org/docs/latest/) (for `main` branch).

### Steps to reproduce the issue

#### I first cloned Open3D by:
```
git clone https://github.com/isl-org/Open3D.git
cd Open3D
```

#### Then, I build Open3D (on Ubuntu 20.04, with CUDA 11.5) with:
```
mkdir build
cd build
cmake -DBUILD_SHARED_LIBS=OFF -DBUILD_CUDA_MODULE=OFF -DUSE_BLAS=ON -DBUILD_WEBRTC=OFF ..
make -j$(nproc)
```

The configure step fails due to an erroneous endif()

### Error message

```shell
CMake Error at CMakeLists.txt:453 (endif):
  Flow control statements are not properly nested.
```

### Open3D, Python and System information

```markdown
Irrelevant in this case
```

### Additional information

_No response_

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -407,7 +407,6 @@ if(BUILD_CUDA_MODULE)
             set(CMAKE_CUDA_FLAGS
                   "${CMAKE_CUDA_FLAGS} -allow-unsupported-compiler -D_ALLOW_COMPILER_AND_STL_VERSION_MISMATCH"
                   CACHE STRING "Flags for NVCC" FORCE)
-            endif()
             message(WARNING "Using --allow-unsupported-compiler flag and -D_ALLOW_COMPILER_AND_STL_VERSION_MISMATCH for nvcc<=12.4 with MSVC>=1940. "
             "Set $Env:NVCC_PREPEND_FLAGS='--allow-unsupported-compiler' if nvcc still fails.")
         endif()

